### PR TITLE
Documenting and cleanup

### LINKED ISSUE #7: ResourceDef tutorial is outdated
I think I changed some variable names, so I might need to check and update.
Mainly putting this here so I don't forget to do that

### PR DESCRIPTION
**Pull request by Toby222**
Add xml-doc comments and format source files

Some places are not be documented because I overlooked them, didn't understand them, or they, too, looked unfinished.
The ones I didn't miss should all have TODOs.

### CODE REVIEW

**Comment by Toby222** [comment]
Gonna need input on code style and stuff as well

File: Source/1.3/AI/AIFacilityManager.cs [hunk 1]
```diff
@@ -50,111 +43,110 @@ public HashSet<FacilityDef> FacilityDefsInstalled
...
             if (settlement != null && manager != null)
             {
-
                 StorageTracker storageTracker = player.Manager.StorageTracker;
                 foreach (ThingDefCountClass thingDefCount in facilityDef.costList)
                 {
-                    hasRemovedAll = hasRemovedAll && storageTracker.CanRemoveThingsFromStorage(thingDefCount.thingDef, thingDefCount.count);
+                    hasRemovedAll = hasRemovedAll &&
+                                    storageTracker.CanRemoveThingsFromStorage(
+                                        thingDefCount.thingDef, thingDefCount.count);
```
Comment (Danimineiro) [inline Source/1.3/AI/AIFacilityManager.cs]: I think these line breaks are really excessive

Comment (Toby222) [reply Source/1.3/AI/AIFacilityManager.cs]: Changed line length to 200

File: Source/1.3/AI/AIFacilityManager.cs [hunk 2]
```diff
@@ -50,111 +43,102 @@ public HashSet<FacilityDef> FacilityDefsInstalled
...
-        /// Attempt to build a facility
+        ///     Attempt to build a facility
         /// </summary>
-        /// <param name="facility"></param>
+        /// <param name="facilityDef"></param>
         /// <returns></returns>
         public bool BuildNewFacility(FacilityDef facilityDef)
         {
-            bool hasRemovedAll = true;
+            var hasRemovedAll = true;
```
Comment (Danimineiro) [inline Source/1.3/AI/AIFacilityManager.cs]: saves a single character => is that really necessary?

File: Source/1.3/Controllers/FactionController/FactionController.cs [hunk 3]
```diff
@@ -1,93 +1,106 @@
...
 
         /// <summary>
-        /// Needed for loading
+        ///     Needed for loading
         /// </summary>
-        public FactionController() { }
+        [UsedImplicitly]
+        public FactionController()
+        {
+        }
```
Comment (Danimineiro) [inline Source/1.3/Controllers/FactionController/FactionController.cs]: Is there a setting to make empty constructors single line?

**Review by Danimineiro — CHANGES_REQUESTED** [review]
Check conversations =>
[constructors single line](https://github.com/Danimineiro/Empire-Rewritten/pull/8/files#r818968290)
[var vs bool](https://github.com/Danimineiro/Empire-Rewritten/pull/8/files#r818954333)

Rest is fine imo

**Comment by Danimineiro** [comment]
Merged master in solving merge conflicts

File: Source/1.3/AI/AIFacilityManager.cs (hunk 2 above)
Comment (Toby222) [reply Source/1.3/AI/AIFacilityManager.cs]: the type is evident, hence the removal. Has nothing to do with length.

Comment (Danimineiro) [reply Source/1.3/AI/AIFacilityManager.cs]: It just doesn't feel very useful in this case. that's what I mean

Comment (Toby222) [reply Source/1.3/AI/AIFacilityManager.cs]: neither option is more useful than the other here, really, but it's consistent this way
(That being said, I'll still check the options)

**Review by RecompiledBirds — APPROVED** [review]

**Review by Danimineiro — APPROVED** [review]

**Pull request merged**